src/server/pps/server/api_server.go

```diff
@@ -1386,6 +1398,15 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *pps.CreatePipel
 				visitErr = err
 			}
 		}
+		if input.Git != nil {
+			if err := pps.ValidateGitCloneURL(input.Git.URL); err != nil {
+				visitErr = err
+				return
+			}
```

Comment:
This should happen in `validateInput` otherwise we might create other input repos and then have to bail out when we realize this one is malformed.